### ISSUE TITLE
Anything called directly from webpack.config.js should be listed as a peer dependency

### ISSUE THREAD

**Issue by QWp6t** [JeffreyWay/laravel-mix#22]
Instead of dependencies, I think these should be peer dependencies. Users have control over the webpack config, and they may opt to use different versions of the loaders, plugins, etc. I think any dependencies not called directly in `src/*` files should be listed as peerDependencies in package.json.

Thoughts?

**Comment by JeffreyWay** [comment]
The user would then have to manually install the loaders, right?

**Comment by QWp6t** [comment]
Unfortunately, yes. But that probably wouldn't be an issue for Laravel since `laravel new` could just include a package.json file that's populated with all of the loaders and whatnot.

But it also occurred to me that the underlying problem I'm trying to solve here (potential version conflicts) is actually probably not likely to be much of a real world issue.

## TL;DR

There are certain edge case scenarios where users would need to specify versions of dependencies to be installed in the topmost node_modules folder. This is probably not actually something to be concerned about, but maybe it should be documented.

## Explanation

Issues can arise if a user specifies a dependency that has its own dependency that conflicts with a dependency specified by laravel-mix. This is because it's not guaranteed which of the dependencies will be in the topmost node_modules folder. I'm not sure how yarn determines which version will be installed in the topmost node_modules folder in such a scenario, but [this blog post explains how npm does it](http://lukecod.es/2015/12/04/npm3-top-level-nested-conflicting-dependencies/). 

Two things to note, though.
1. This isn't actually an issue unless the user is calling the topmost dependency (such as `webpack` or one of its plugins/loaders/etc).
2. This issue is resolved when the user specifies their required version of the dependency in their package.json file.

## Example

`laravel-mix` uses webpack@2, but a user adds another dependency that lists webpack@1 (or in the future they might list webpack@3) as a dependency. It's not known with certainty which version of webpack will be installed in the topmost node_modules folder at this point. The user can resolve the issue by adding webpack@2 as a dependency.

## Conclusion

So maybe the best solution here is to stick with the convenience of having them listed as dependencies, but perhaps add a note about the potential edge case issues in the laravel-mix docs.

**Comment by JeffreyWay** [comment]
Yeah good point. Thanks for the explanation.

If there was a way to do it... that didn't involve giving folks some long `npm install` command to run, I'd be okay with it. But, ideally, they do `npm install laravel-mix` and they're done.

Thanks again!
